### ISSUE TITLE
After editing mentions in a post, it shows `@[deleted]`.

### ISSUE THREAD

**Issue by rafaucau** [flarum/core#3074]
## Bug Report

**Current Behavior**
When I edit / add mentions in a post, it shows `@[deleted]` after saving. Refreshing the page helps.

**Steps to Reproduce**
1. Mention someone in the post
2. Edit this post to add another mention or edit a previous mention
3. Save changes

**Expected Behavior**
Mentions should display correctly after saving an edit without having to refresh the page.

**Screenshots**

https://user-images.githubusercontent.com/25438601/133886430-98301aa9-089e-47ba-af84-2317c7da2950.mp4

https://user-images.githubusercontent.com/25438601/133886492-f3f58c63-bf29-479a-b3c3-b812bea6f209.mp4

**Environment**
- Flarum version: 1.0.4

- Webserver: [apache, litespeed]
- Hosting environment: [local, shared]
- PHP version: [7.4, 8.0]
- Browser: Chrome 93 but probably all

```
Output of "php flarum info", run this in terminal in your Flarum directory.

Flarum core 1.0.4
PHP version: 8.0.3
Loaded extensions: Core, bcmath, calendar, ctype, date, filter, hash, iconv, json, SPL, pcre, readline, Reflection, session, standard, mysqlnd, tokenizer, zip, zlib, libxml, dom, PDO, bz2, SimpleXML, xml, xmlreader, xmlwriter, openssl, curl, fileinfo, gd, gettext, gmp, intl, imap, ldap, mbstring, exif, mysqli, Phar, pdo_mysql, pdo_sqlite, soap, sockets, xsl
+-----------------------------------+---------+--------+
| Flarum Extensions                 |         |        |
+-----------------------------------+---------+--------+
| ID                                | Version | Commit |
+-----------------------------------+---------+--------+
| flarum-flags                      | v1.0.0  |        |
| flarum-tags                       | v1.0.3  |        |
| clarkwinkelmann-create-user-modal | 1.3.0   |        |
| flarum-suspend                    | v1.0.0  |        |
| flarum-markdown                   | v1.0.1  |        |
| flarum-subscriptions              | v1.0.0  |        |
| fof-formatting                    | 1.0.1   |        |
| flarum-approval                   | v1.0.0  |        |
| flarum-nicknames                  | v1.0.0  |        |
| flarum-sticky                     | v1.0.0  |        |
| flarum-statistics                 | v1.0.0  |        |
| flarum-mentions                   | v1.0.0  |        |
| flarum-likes                      | v1.0.0  |        |
| flarum-emoji                      | v1.0.0  |        |
| flarum-bbcode                     | v1.0.0  |        |
| rob006-lang-polish                | v0.5.9  |        |
| flarum-lang-english               | v1.0.0  |        |
| migratetoflarum-fake-data         | 1.0.1   |        |
| flarum-lock                       | v1.0.0  |        |
| fof-upload                        | 1.0.2   |        |
+-----------------------------------+---------+--------+
Base URL: http://flarum.localhost
Installation path: D:\Serwer\wamp\www\flarum
Debug mode: ON
```

**Comment by davwheat** [comment]
I think this is a duplicate of #2989

**Comment by SychO9** [comment]
Yup, Duplicate of #2989, has been fixed for 1.1
